backend/src/index.ts

```
@@ -159,14 +173,16 @@ export function applyRateLimit(limitOverride?: number) {
       resetAt = current.resetAt;
     }
 
-    res.setHeader("X-RateLimit-Limit", String(maxRequests));
-    res.setHeader("X-RateLimit-Remaining", String(Math.max(0, maxRequests - count)));
-    res.setHeader("X-RateLimit-Reset", String(Math.ceil(resetAt / 1000)));
+    res.setHeader('X-RateLimit-Limit', String(maxRequests));
+    res.setHeader('X-RateLimit-Remaining', String(Math.max(0, maxRequests - count)));
+    res.setHeader('X-RateLimit-Reset', String(Math.ceil(resetAt / 1000)));
+
+    console.log(`[RateLimit] key=${key} current=${JSON.stringify(current)} count=${count} maxRequests=${maxRequests} shouldThrow=${current && now < current.resetAt && current.count >= maxRequests}`);
```

ESLint

[error] Unexpected console statement.

(no-console)